tasks/human_eval_129.rs

```diff
@@ -9,10 +9,450 @@ use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+// specification
+pub closed spec fn path_less_than(path1: Seq<int>, path2: Seq<int>) -> bool
+    recommends
+        path1.len() == path2.len(),
+{
+    (exists|n: int|
+        (0 <= n < path1.len() && path1[n] < path2[n] && (forall|i: int|
+            0 <= i < n ==> path1[i] == path2[i])))
+}
+
+pub closed spec fn is_adjacent<const N: usize>(
+    grid: Seq<Seq<int>>,
+    m: int,
+    n: int,
+    coords: (int, int),
+) -> bool
+    recommends
+        grid.len() == N,
+        forall|i: int| (0 <= i < N ==> #[trigger] grid[i].len() == N),
+{
+    &&& (0 <= coords.0 <= N - 1)
+    &&& (0 <= coords.1 <= N - 1)
+    &&& (grid[coords.0][coords.1]) == m
+    &&& ((coords.0 < N - 1 && grid[coords.0 + 1][coords.1] == n) || (coords.0 > 0
+        && grid[coords.0][coords.1] == m && grid[coords.0 - 1][coords.1] == n) || (coords.1 < N - 1
+        && grid[coords.0][coords.1] == m && grid[coords.0][coords.1 + 1] == n) || (coords.1 > 0
+        && grid[coords.0][coords.1] == m && grid[coords.0][coords.1 - 1] == n))
+}
+
+pub closed spec fn adjacent_numbers<const N: usize>(
+    grid: Seq<Seq<int>>,
+    m: int,
+    n: int,
+) -> bool  // TODO : will have to make these u8's into ints
+
+    recommends
+        grid.len() == N,
+        forall|i: int| (0 <= i < N ==> #[trigger] grid[i].len() == N),
+{
+    exists|i: int, j: int| #[trigger] (is_adjacent::<N>(grid, m, n, (i, j)))
+}
+
+pub closed spec fn exists_a_square_with_value<const N: usize>(grid: Seq<Seq<int>>, n: int) -> bool
```

Comment:
using `cell` instead of `square` might be clearer here

Reply:
done